src/MySqlConnector/MySqlClient/Caches/CachedParameter.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Data;
+using System.Linq;
+using MySql.Data.MySqlClient.Types;
+
+namespace MySql.Data.MySqlClient.Caches
+{
+	internal class CachedParameter
+	{
+		public CachedParameter(int ordinalPosition, string mode, string name, string dataType, bool unsigned)
+		{
+			Position = ordinalPosition;
+			if (Position == 0)
+			{
+				Direction = ParameterDirection.ReturnValue;
+			}
+			else
+			{
+				switch (mode.ToLower())
```

Comment:
Need to use `ToLowerInvariant` to avoid the "[Turkish I problem](http://haacked.com/archive/2012/07/05/turkish-i-problem-and-why-you-should-care.aspx/)".